geonode/upload/views.py

```diff
@@ -415,7 +415,7 @@ def check_step_view(request, upload_session):
                     upload_session.completed_step = 'check'
                 else:
                     # This command skip completely 'time' configuration
-                    upload_session.completed_step = 'time' if _ALLOW_TIME_STEP else 'run'
+                    upload_session.completed_step = 'time' if _ALLOW_TIME_STEP else 'check'
```

Comment:
This looks like an infinite recursive loop.

Reply:
It's not an infinite recursive loop (completed step is set to 'check', so the next step will be taken), so for shape-files, the next step that will be executed would be 'run'. With the current code in master, it would say 'run' is completed already, so it would directly go to final (which might be the correct behavior). None-the-less, I don't see why the uncorrect store is being used (since this is set in the check-step, I assumed this was not executed, yet apparently it is)

I will analyse this further, thanks for your efforts